### ISSUE TITLE
Inconsistent results with Unity project.

### ISSUE THREAD

**Issue by mikew** [OmniSharp/omnisharp-atom#510]
Two machines, both new installs of OSX, Unity, and Atom, both sets of packages installed the same way.

We've run `rm *.sln *.csproj` on both machines, then use the "Open C# Project" in Unity's menu (formerly "Sync MonoDevelop Project".

At this point, on one machine, everything works a charm. On the other machine, the second I go to edit any code the log fills up with errors about missing references, even to plenty of built in stuff.

There is no difference in the `.sln` files, and the only difference in the `*.csproj` files is a full path to a DLL and some Unity defines (working machine has a pro license and the other machine is a personal license).

Has anyone come across this?

**Comment by nosami** [comment]
Can you open the solution that doesn't work in MonoDevelop or Xamarin Studio?

**Comment by mikew** [comment]
Yep, and everything works as expected in MonoDevelop.

**Comment by david-driscoll** [comment]
Do you have a sample of the errors you're seeing?
On Thu, Sep 17, 2015 at 11:59 AM mikew <notifications@github.com> wrote:

> Yep, and everything works as expected in MonoDevelop.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OmniSharp/omnisharp-atom/issues/510#issuecomment-141132282>
> .
>
-- 
Thanks,
David

**Comment by mikew** [comment]
It goes between not being able to find UnityEngine stuff, to finding that but failing half-way through loading the projects code.

https://gist.github.com/mikew/319d3d47339c53e98b15#file-omnisharp-output-log is log from omnisharp. Some interesting bits:

[Line 108](https://gist.github.com/mikew/319d3d47339c53e98b15#file-omnisharp-output-log):
```
An unhandled exception has occurred: Diagnostic must have span '[1403..1403)'
```

[Line 70](https://gist.github.com/mikew/319d3d47339c53e98b15#file-omnisharp-output-log-L70)
```
Failed to process project file '/Users/pankaj/src/alaska/Assembly-CSharp-Editor.csproj'
```

**Comment by nosami** [comment]
@mikew what are the names of the csproj files? Unity used to emit 2 out per project and one used to end in -csharp.csproj. That's the one that we use for Unity.

Maybe they changed something?

**Comment by nosami** [comment]
@mikew see here https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp/MSBuild/SolutionPicker.cs#L17

**Comment by mikew** [comment]
Yeah it looks like Unity 5.2.0 drops the `*-csharp.sln` file. All of the `*-vs.csproj` are gone as well. [This blog post](http://blogs.unity3d.com/2015/09/08/unity-5-2-easy-access-to-unity-services/) mentions "Native Visual Studio Integration".

Either way, neither machine has any `*-csharp.sln` file, but we are still in this jam.

**Comment by david-driscoll** [comment]
I wonder if you're able to drop back to something pre 5.2.0, and see if it works there?  It could be need some sort of alternative if those files are gone.

**Comment by mikew** [comment]
Once work is finished I'll try 5.1 and 5.2, with and without licenses, in a project I can share the source in, and see what happens.

**Comment by mikew** [comment]
The issue persisted with 5.1.3, with either Pro or Personal licenses.

The issue **did not** crop up on a new project, everything worked fine there.

I tried opening the project in Vim with YouCompleteMe, and it could load the references omnisharp-atom said it couldn't find.

![atom](https://cloud.githubusercontent.com/assets/4729/9948825/ee322080-5d7c-11e5-9604-fcbdc93abc42.png)
![vim](https://cloud.githubusercontent.com/assets/4729/9948826/ee34fabc-5d7c-11e5-9f20-1d9fe0a6fa08.png)

**Comment by nosami** [comment]
YouCompleteMe uses the old version of the server (omnisharp-server vs omnisharp-roslyn)

Is it possible for you to share your code? maybe privately?

**Comment by mikew** [comment]
Sure, hit my email, it's on my profile.

**Comment by nosami** [comment]
Emailed!

**Comment by nosami** [comment]
Seems to be working for me.

<img width="685" alt="screen shot 2015-09-18 at 09 44 01" src="https://cloud.githubusercontent.com/assets/667194/9955563/e0bc6f2e-5de9-11e5-896f-5e14e649896d.png">

**Comment by mikew** [comment]
Just as I feared. Thanks for taking a look tho!

**Comment by nosami** [comment]
On the machine that doesn't work, can you just check the references in your csproj, and see if they exist on disk where they are supposed to be?

**Comment by Angrycrow** [comment]
How do I find the log file? I've been having a similar issue. Where. In the event that I have to force-quit ( usually from Cutting or Deleting multiple lines over a certain size * that cause a temporary error * . Unclosed if  as an example ).  Atom will hang. I know it's an omnisharp issue , or I've narrowed it down that far at least.  Would be fine, except for when I reload the file. The internal ( atom? ) backup replaces the current working version from unity. This has resulted re-fixing a project more than once. How can I help this ?

**Comment by david-driscoll** [comment]
@Angrycrow For that hanging issue, do you have Enhanced Highlighting on?  There is an outstanding issue, that appears to cause atom to get stuck in an infinite loop.  Disabling Enhanced Highlighting has been known to fix it (it is also the reason that enhanced highlighting is off by default still).

I've not been able to get the Chromium Debug tools to get in there while the issue is happening to determine what the real and true fix is.  If you can some extra traction in there that would be helpful, but you certainly don't have to.

**Comment by Angrycrow** [comment]
oh.. While the issue is happening.. What am I looking for ? 
<img width="1440" alt="screen shot 2015-09-21 at 12 43 29 pm" src="https://cloud.githubusercontent.com/assets/2450703/9999889/78942474-605e-11e5-9ec2-28ad8f6bc964.png">

**Comment by david-driscoll** [comment]
From what I've seen, as I've been unable to narrow it down, it has to do
with something in the atom core code, around tokenization.  It's also
possible it's something OmniSharp-atom is doing, that atom just doesn't
like.  Unfortunately I don't have much more than I've seen it happen, and
turning off highlighting has been the fix.

**Comment by Angrycrow** [comment]
hermm...  This is worth learning more about.. Let's see.. I'll tinker around and see if I can re-create it in these conditions.

**Comment by Angrycrow** [comment]
I think I caught it while capturing , But I don't know if it'll ever finish retrieving events... hanging on this now :
<img width="1440" alt="screen shot 2015-09-21 at 1 05 55 pm" src="https://cloud.githubusercontent.com/assets/2450703/10000411/954ee718-6061-11e5-9a5c-235287bf3e2a.png">

**Comment by Angrycrow** [comment]
I've been able to re-create this condition pretty reliably. Let me know if you need the solution to see what happens. All I've been doing is cutting out the condition at the end of that method.        
```c#
 // draggin detail.
        if ( isDragging )
        {
            arm.SetActive(false);
            drag.SetActive(true);
        }
        else
        {
            drag.SetActive(false);
        }
```

**Comment by david-driscoll** [comment]
If you have the repo somewhere that'd be great, if it's closed source, I
can get you my email, and I can test it there.

**Comment by Angrycrow** [comment]
Here's a link to the project ( Unity 5.1 ),  let me know if you can recreate the bug. I'm runnin OSX 10.10.5 and the current atom/omniSharp release. 

https://www.dropbox.com/s/06fsar9383tkifr/Archive_TROLL_LD33.zip?dl=1

**Comment by david-driscoll** [comment]
I think I finally nailed this... hopefully this is the last edge case we have...

**Comment by david-driscoll** [comment]
@Angrycrow try v0.24.13 and let me know if the problem is gone for you (it was for me with your code anyway)

**Comment by Angrycrow** [comment]
Checked the pre : Issue still exists. 
*updating*
Sorry to say. Still getting it. Disabling that green pencil icon thing certainly works around it. But I was able to recreate the bug just the same:

<img width="1440" alt="screen shot 2015-09-24 at 10 12 49 pm" src="https://cloud.githubusercontent.com/assets/2450703/10092204/84020d78-6309-11e5-8e92-2c37205122e3.png">

**Comment by david-driscoll** [comment]
on quick question did you restart atom after you updated?

**Comment by Angrycrow** [comment]
I did. Yes. I'm re-installing omniSharp just to be sure. Atom is V 1.0.18 currently?

**Comment by Angrycrow** [comment]
I found another clue. Trying to nail down the specific behavior. 
so in  . 
``` cs
if ( isDragging )
        {
            arm.SetActive(false);
            drag.SetActive(true);
        }
        else
        {
            drag.SetActive(false);
        }
```

**Comment by david-driscoll** [comment]
I see it now...

**Comment by david-driscoll** [comment]
I _think_ I found the common edge cases (I'm sure there are more).

@Angrycrow you mind trying out this branch locally https://github.com/OmniSharp/omnisharp-atom/pull/522 and letting me know if it works for you? (If you need instructions just reply or hit me up on Jabbr ( https://jabbr.net/#/rooms/omnisharp )
